### PR TITLE
change license uploading to parse RHSM manifests

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

File: awx/api/views/root.py [hunk 1]
```diff
@@ -350,22 +351,25 @@ def post(self, request):
...
-            from awx.main.utils.common import get_licenser
-            license_data_validated = get_licenser().validate(new_cert=True)
-        except Exception:
-            logger.warning(smart_text(u"Invalid license submitted."),
-                           extra=dict(actor=request.user.username))
-            # If License invalid, clear entitlment cert value
-            settings.ENTITLEMENT_CERT = ''
-            return Response({"error": _("Invalid License")}, status=status.HTTP_400_BAD_REQUEST)
+            try:
+                # Validate entitlement cert and get subscription metadata
```
Comment (wenottingham) [inline awx/api/views/root.py:364]: super picky: comments don't reflect what we're now doing

Comment (ryanpetrello) [reply awx/api/views/root.py:364]: correct, this PR needs a lot of love and nitpickiness; for now we're settling on "does it work"

Comment (ryanpetrello) [reply awx/api/views/root.py:364]: I'd like to take the *entire* final PR, squash it, and aggressively clean up cruft and write tests.

File: awx/main/utils/licensing.py [hunk 2]
```diff
@@ -40,6 +47,26 @@ def rhsm_config():
...
+    sig = z.open('signature').read()
+    with open('/etc/tower/certs/candlepin-redhat-ca.crt', 'rb') as f:
+        cert = x509.load_pem_x509_certificate(f.read(), backend=default_backend())
+        key = cert.public_key()
+    key.verify(sig, export, padding=padding.PKCS1v15(), algorithm=hashes.SHA256())
+
+    buff.write(export)
+    z = zipfile.ZipFile(buff)
+    for f in z.filelist:
+        if f.filename.startswith('export/entitlements') and f.filename.endswith('.json'):
```
Comment (wenottingham) [inline awx/main/utils/licensing.py:66]: I would ask the subs team whether requiring they only add one sub to the manifest will work in the future

File: awx/ui/client/src/license/license.controller.js [hunk 3]
```diff
@@ -111,14 +111,14 @@ export default
             const raw = new FileReader();
 
             raw.onload = function() {
-                $scope.newLicense.entitlement_cert = raw.result;
+                $scope.newLicense.manifest = btoa(raw.result);
```
Comment (ryanpetrello) [inline awx/ui/client/src/license/license.controller.js:114]: There's probably a better way to do this correctly, but it likely involves changing our POST content-type to have binary data in the content body *or* JSON and meeeeeeeeeeeeeh whatever this works fine.

Comment (ryanpetrello) [reply awx/ui/client/src/license/license.controller.js:114]: cc @mabashian

**Pull request closed without merging**